static/docs/commands-reference/commit.md

```diff
@@ -2,40 +2,212 @@
 
 Record changes to the repository by saving outputs to cache.
 
+## Synopsis
+
 ```usage
     usage: dvc commit [-h] [-q | -v] [-f] [-d] [-R] [targets [targets ...]]
 
     positional arguments:
       targets               DVC files.
-
-    optional arguments:
-      -f, --force           Commit even if checksums for dependencies/outputs
-                            changed.
-      -d, --with-deps       Commit all dependencies of the specified target.
-      -R, --recursive       Commit cache for subdirectories of the specified
-                            directory.
 ```
 
+## Description
+
+Normally DVC commands like `dvc add`, `dvc repro` or `dvc run`, commit the data
+to the DVC cache as the last step. What _commit_ means is that DVC
+
+* Computes a checksum for the file
+* Enters the checksum and file name into the DVC stage file
+* Tells the SCM to ignore the file (e.g. add entry to `.gitignore`).  If the
+  workspace was initialized with no SCM support (`dvc init --no-scm`) this does
+  not happen.
+* Adds the file to the DVC cache
+
+There are many cases where the last step is not desirable. For the DVC commands
+where it is appropriate the `--no-commit` option prevents the last step from
+occurring. The checksum is still computed and added to the DVC file, but the
+file is not added to the cache.
+
+That's where the `dvc commit` command comes into play. It handles that last
```

Comment:
It's not exactly true, actually - `dvc commit` is not an exact opposite of `--no-commit`. The first one is updating DVC files + saves data to cache. Second one is still updating files but do not saves to cache. So, to be precise - `dvc commit` is more than just the last step. So, for example you can edit some data file by hand and run `dvc commit` to update DVC file. It's useful in the `dvc add` -> update file/add or remove files from dir -> `dvc commit`.

Reply:
With --no-commit the DVC files are updated, but the data files are not added to the cache.  

As for "you can edit some data file" .. I am under the impression that one must first `dvc unprotect` or `dvc remove` the file to avoid polluting the cache.  Hence the examples I've written at the bottom.

Reply:
Yep. We still need to clarify this difference. It's important that `dvc commit` is not just an opposite of the `--no-commit`. It can be used in other scenarios such as manual/outside of DVC data changes (yes, you need to unprotect the files in this case, I would mention a link to the article about "unprotecting and editing files", but would not emphasize it too much here).

Reply:
I think I have it now